frontend/javascripts/oxalis/view/settings/dataset_settings_view.js

```diff
@@ -76,9 +78,76 @@ type DatasetSettingsProps = {|
   onChangeUser: (key: $Keys<UserConfiguration>, value: any) => void,
   onUnlinkFallbackLayer: Tracing => Promise<void>,
   tracing: Tracing,
+  task: ?Task,
 |};
 
-class DatasetSettings extends React.PureComponent<DatasetSettingsProps> {
+function DownsampleVolumeModal({ visible, hideDownsampleVolumeModal, magsToDownsample }) {
+  const [isDownsampling, setIsDownsampling] = useState(false);
+
+  const handleTriggerDownsampling = async () => {
+    setIsDownsampling(true);
+    await api.tracing.downsampleSegmentation();
+    setIsDownsampling(false);
+  };
+
+  return (
+    <Modal
+      title="Downsample Volume Annotation"
+      onCancel={isDownsampling ? null : hideDownsampleVolumeModal}
+      visible={visible}
+      footer={null}
+      width={800}
+      maskClosable={false}
+    >
+      <p>
+        This annotation does not have volume annotation data in all resolutions. Consequently,
+        annotation data cannot be rendered at all zoom values. By clicking &quot;Downsample&quot;,
+        webKnossos will use the best resolution of the volume data to create all dependent
+        resolutions.
```

Comment:
This may overwrite existing resolution, for example, if mags 1 and 2 exist, mag 2 is overwritten, right? If so, this should be noted here or maybe better, in the next paragraph: "The following resolutions will be overwritten ...".

Reply:
If my previous comment holds true, is there a technical reason why the best resolution is used for downsampling, possibly overwriting existing resolutions, instead of downsampling from the "worst" existing resolution?

Reply:
Technically, you are correct, but conceptionally the mags should always be in sync. Thus, discarding downsampled mags and re-creating them should not do any harm, which is why I wouldn't want to unsettle/confuse the users with that information. Do you agree?

Reply:
> Technically, you are correct, but conceptionally the mags should always be in sync.

Yes if that's the case, I agree :)